src/main/java/core/basesyntax/MyLinkedList.java

```diff
@@ -3,48 +3,176 @@
+
+        public Node(Node<T> prev, T item, Node<T> next) {
+            this.item = item;
+            this.next = next;
+            this.prev = prev;
+        }
+    }
+
+    private Node<T> first;
+    private Node<T> last;
+    private int size;
+
+    public MyLinkedList() {
+        size = 0;
+    }
+
     @Override
     public boolean add(T value) {
-        return false;
+        if (size == 0) {
+            first = new Node<>(null, value, null);
+            last = first;
+            size++;
+            return true;
+        }
+        Node<T> preLast = last;
+        preLast.next = new Node<>(preLast, value, null);
+        last = preLast.next;
+        size++;
+        return true;
     }
 
     @Override
     public void add(T value, int index) {
-
+        if ((isEmpty() && index == 0) || index == size) {
+            add(value);
+        } else if (index == 0) {
+            Node<T> add = new Node<>(null, value, first);
+            first.prev = add;
+            first = add;
+            size++;
+        } else {
+            validateIndex(index);
+            Node<T> node = getNode(index);
+            Node<T> add = new Node<>(node.prev, value, node);
+            node.prev.next = add;
+            node.prev = add;
+            size++;
+        }
     }
 
     @Override
     public boolean addAll(List<T> list) {
-        return false;
+        for (T t : list) {
+            add(t);
+        }
+        return true;
     }
 
     @Override
     public T get(int index) {
-        return null;
+        return getNode(index).item;
     }
 
     @Override
     public T set(T value, int index) {
-        return null;
+        Node<T> node = getNode(index);
+        T removed = node.item;
+        node.item = value;
+        return removed;
     }
 
     @Override
     public T remove(int index) {
-        return null;
+        Node<T> remove = getNode(index);
+        removeNode(remove);
+        return remove.item;
     }
 
     @Override
     public boolean remove(T object) {
-        return false;
+        if (isEmpty()) {
+            return true;
+        } else if (object == null) {
+            for (Node<T> node = first; node != null; node = node.next) {
+                if (null == node.item) {
+                    removeNode(node);
+                    return true;
+                }
+            }
+            return false;
+        } else {
+            for (Node<T> node = first; node != null; node = node.next) {
+                if (object.equals(node.item)) {
+                    removeNode(node);
+                    return true;
+                }
+            }
+            return false;
+        }
     }
 
     @Override
     public int size() {
-        return 0;
+        return size;
     }
 
     @Override
     public boolean isEmpty() {
-        return false;
+        return first == null;
+    }
+
+    public void removeFirst() {
+        if (size == 1) {
+            T item = first.item;
```

Comment:
do you use this variable?

Reply:
At the beginning I meant to return item I remove, but... sure, I removed it